content/projects/gsoc/2020/project-ideas/jenkins-distribution-customize-service.adoc

```diff
@@ -0,0 +1,41 @@
+---
+layout: gsocprojectidea
+title: "Jenkins distribution customize service"
+goal: "Provide an out of the box solution for Jenkins users"
+category: Service
+year: 2020
+status: draft
+skills:
+- Java
+- Web Development
+mentors:
+- "linuxsuren"
+---
+
+For many users, they download Jenkins first, then select some plugins and config them. 
+It might take a lot of time, like hours. But if we can get a perfect Jenkins distribution which contains all we need, 
+it can save that time for us.
+
+The service could be in the form of  a website offered as https://customize.jenkins.io. Plus, it should support self-host. 
+People can select the following configuration items:
+
+* Jenkins core version
+* plugins
+* common configuration, user/password, update-center site e.g.
```

Comment:
How? JCasC? Or would it be dialog-based?

Reply:
I'd prefer it's scenario-based instead of using JCasC directly. I hope it can be very easy to use. Even for some new users, they don't need to know too much about Jenkins to start it.

Reply:
If we do that, it would be great to share the implementation with the Jenkins Installation Wizard

Reply:
+1 from me